subxt/src/backend/legacy/mod.rs

```diff
@@ -84,183 +85,249 @@ impl<T: Config + Send + Sync + 'static> Backend<T> for LegacyBackend<T> {
         keys: Vec<Vec<u8>>,
         at: T::Hash,
     ) -> Result<StreamOfResults<StorageResponse>, Error> {
-        let methods = self.methods.clone();
+        async fn fut<T: Config>(
+            keys: Vec<Vec<u8>>,
+            at: T::Hash,
+            methods: LegacyRpcMethods<T>,
+        ) -> Result<StreamOfResults<StorageResponse>, Error> {
+            // For each key, return it + a future to get the result.
+            let iter = keys.into_iter().map(move |key| {
+                let methods = methods.clone();
+                async move {
+                    let res = methods.state_get_storage(&key, Some(at)).await?;
+                    Ok(res.map(|value| StorageResponse { key, value }))
+                }
+            });
 
-        // For each key, return it + a future to get the result.
-        let iter = keys.into_iter().map(move |key| {
-            let methods = methods.clone();
-            async move {
-                let res = methods.state_get_storage(&key, Some(at)).await?;
-                Ok(res.map(|value| StorageResponse { key, value }))
-            }
-        });
+            let s = stream::iter(iter)
+                // Resolve the future
+                .then(|fut| fut)
+                // Filter any Options out (ie if we didn't find a value at some key we return nothing for it).
+                .filter_map(|r| future::ready(r.transpose()));
 
-        let s = stream::iter(iter)
-            // Resolve the future
-            .then(|fut| fut)
-            // Filter any Options out (ie if we didn't find a value at some key we return nothing for it).
-            .filter_map(|r| future::ready(r.transpose()));
+            Ok(StreamOf(Box::pin(s)))
+        }
 
-        Ok(StreamOf(Box::pin(s)))
+        retry(|| fut(keys.clone(), at, self.methods.clone())).await
     }
 
     async fn storage_fetch_descendant_keys(
         &self,
         key: Vec<u8>,
         at: T::Hash,
     ) -> Result<StreamOfResults<Vec<u8>>, Error> {
-        let keys = StorageFetchDescendantKeysStream {
-            at,
-            key,
-            storage_page_size: self.storage_page_size,
-            methods: self.methods.clone(),
-            done: Default::default(),
-            keys_fut: Default::default(),
-            pagination_start_key: None,
-        };
+        retry(|| async {
```

Comment:
Is this retry doing anything? there is no `.await` inside it so AFAIU it's just returning the storage stream and will never retry?